packages/arb-node-core/nodehealth/nodehealth.go

```diff
@@ -200,38 +199,18 @@ func nodeReadinessChecks(health healthcheck.Handler, httpMux *http.ServeMux, aSy
 	httpMux.HandleFunc("/ready", health.ReadyEndpoint)
 }
 
-//Define which healthchecks to use for the liveness API and expose the liveness API
-func nodeLivenessChecks(health healthcheck.Handler, httpMux *http.ServeMux) {
-	//Create an endpoint to serve the liveness check
-	httpMux.HandleFunc("/live", health.LiveEndpoint)
-}
-
-//Expose the prometheus metrics API along with the raw responses from OpenEthereum
-func nodeMetrics(httpMux *http.ServeMux, prometheusRegistry *prometheus.Registry) {
-	//Create an endpoint to serve the prometheus endpoint
-	httpMux.Handle("/metrics", promhttp.HandlerFor(
-		prometheusRegistry, promhttp.HandlerOpts{}))
-}
-
-//Create the HTTP server and start a watchdog to monitor its return codes
-func healthcheckServer(httpMux *http.ServeMux, config configStruct) {
-	http.ListenAndServe(config.healthcheckRPC, httpMux)
-}
-
 func waitConfig(config *configStruct) {
 	config.mu.Lock()
+	defer config.mu.Unlock()
 	for config.openethereumHealthcheckRPC == "" {
-		config.mu.Unlock()
 		time.Sleep(config.loopDelayTimer)
```

Comment:
Probably want to avoid holding the lock over the sleep. In general I'd say using the `Lock` and defer `Unlock` is better unless there's something slow in the function in which case making it more fine-grain is better